### PR TITLE
Expose S3 and Azure contructors

### PR DESCRIPTION
**Pull request by skejserjensen**
Multiple types support operating on data in the local file system and remote object stores. However, arguments had to be serialized to bytes with a homemade encoding and passed to `try_from_connection_info()` as a `&[u8]` to use these types with a remote object store. This PR add `try_from_s3_configuration()` and `try_from_azure_configuration()` to simplify using the types with a remote object store. While implementing these methods, unnecessary use of `async` was found and removed, and methods operating on local paths where changed from `&str` to `std::path::Path` to make it clear that a local path must be passed. To differentiate between `std::path::Path` and `object_store::path::Path` and keep the code consistent is `std::path::Path` always imported as `std::path::Path as StdPath` even if  `object_store::path::Path` is not in the same scope.

### CODE REVIEW

**Review by chrthomsen — APPROVED** [review]

File: crates/modelardb_common/src/storage.rs [hunk 1]
```diff
@@ -77,62 +78,107 @@ impl DeltaLake {
                 .map_err(|error| DeltaTableError::generic(error.to_string()))?,
         );
 
+        let location = data_folder_path
+            .to_str()
+            .ok_or_else(|| DeltaTableError::generic("Local data folder path is not UTF-8"))?
```
Comment (CGodiksen) [inline crates/modelardb_common/src/storage.rs:83]:
```suggestion
            .ok_or_else(|| DeltaTableError::generic("Local data folder path is not UTF-8."))?
```

File: crates/modelardb_common/src/storage.rs [hunk 2]
```diff
@@ -77,62 +78,107 @@ impl DeltaLake {
...
+                )
             }
             _ => Err(DeltaTableError::Generic(format!(
                 "{object_store_type} is not supported."
             ))),
-        }?;
+        }
+    }
+
+    /// Create a new [`DeltaLake`] that manages the delta tables in an object store with an
```
Comment (CGodiksen) [inline crates/modelardb_common/src/storage.rs:133]: Minor but should it be `Delta tables`?

File: crates/modelardb_common/src/metadata/mod.rs [hunk 3]
```diff
@@ -52,77 +53,111 @@ pub struct MetadataDeltaLake {
 impl MetadataDeltaLake {
     /// Create a new [`MetadataDeltaLake`] that saves the metadata to [`METADATA_FOLDER`] under
     /// `folder_path`.
-    pub fn from_path(folder_path: &str) -> MetadataDeltaLake {
-        MetadataDeltaLake {
-            url_scheme: format!("{folder_path}/{METADATA_FOLDER}"),
+    pub fn from_path(folder_path: &StdPath) -> Result<MetadataDeltaLake, DeltaTableError> {
+        let folder_str = folder_path
+            .to_str()
+            .ok_or_else(|| DeltaTableError::generic("Local data folder path is not UTF-8"))?;
```
Comment (CGodiksen) [inline crates/modelardb_common/src/metadata/mod.rs:59]:
```suggestion
            .ok_or_else(|| DeltaTableError::generic("Local data folder path is not UTF-8."))?;
```

File: crates/modelardb_common/src/metadata/mod.rs [hunk 4]
```diff
@@ -52,77 +53,111 @@ pub struct MetadataDeltaLake {
...
             }
             _ => Err(DeltaTableError::Generic(format!(
                 "{object_store_type} is not supported."
             ))),
-        }?;
+        }
+    }
+
+    /// Create a new [`MetadataDeltaLake`] that saves the metadata to [`METADATA_FOLDER`] in a
+    /// remote S3 object store. If a connection cannot be created [`DeltaTableError`] is returned.
```
Comment (CGodiksen) [inline crates/modelardb_common/src/metadata/mod.rs:108]: Maybe change to "S3-compatible".

File: crates/modelardb_common/src/metadata/mod.rs [hunk 5]
```diff
@@ -52,77 +53,111 @@ pub struct MetadataDeltaLake {
...
+
+        Ok(MetadataDeltaLake {
+            location,
+            storage_options,
+            session: SessionContext::new(),
+        })
+    }
+
+    /// Create a new [`MetadataDeltaLake`] that saves the metadata to [`METADATA_FOLDER`] in a
+    /// remote Azure object store. If a connection cannot be created [`DeltaTableError`] is
```
Comment (CGodiksen) [inline crates/modelardb_common/src/metadata/mod.rs:136]: Maybe change to "Azure-compatible".

File: crates/modelardb_common/src/metadata/table_metadata_manager.rs [hunk 6]
```diff
@@ -76,11 +77,61 @@ impl TableMetadataManager {
...
+        };
+
+        table_metadata_manager
+            .create_metadata_delta_lake_tables()
+            .await?;
+
+        Ok(table_metadata_manager)
+    }
+
+    /// Create a new [`TableMetadataManager`] that saves the metadata to a remote S3 object store
```
Comment (CGodiksen) [inline crates/modelardb_common/src/metadata/table_metadata_manager.rs:95]: Maybe change to "S3-compatible".

File: crates/modelardb_common/src/metadata/table_metadata_manager.rs [hunk 7]
```diff
@@ -76,11 +77,61 @@ impl TableMetadataManager {
...
+        };
+
+        table_metadata_manager
+            .create_metadata_delta_lake_tables()
+            .await?;
+
+        Ok(table_metadata_manager)
+    }
+
+    /// Create a new [`TableMetadataManager`] that saves the metadata to a remote Azure object store
```
Comment (CGodiksen) [inline crates/modelardb_common/src/metadata/table_metadata_manager.rs:121]: Maybe change to "Azure-compatible".

**Review by CGodiksen — APPROVED** [review]

File: crates/modelardb_common/src/metadata/mod.rs (hunk 3 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/metadata/mod.rs:59]: Fixed.

File: crates/modelardb_common/src/storage.rs (hunk 1 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/storage.rs:83]: Fixed.

File: crates/modelardb_common/src/storage.rs (hunk 2 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/storage.rs:133]: Fixed everywhere in repository for consistentcy.

File: crates/modelardb_common/src/metadata/mod.rs (hunk 5 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/metadata/mod.rs:136]: Fixed.

File: crates/modelardb_common/src/metadata/table_metadata_manager.rs (hunk 6 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/metadata/table_metadata_manager.rs:95]: Fixed.

File: crates/modelardb_common/src/metadata/table_metadata_manager.rs (hunk 7 above)
Comment (skejserjensen) [reply crates/modelardb_common/src/metadata/table_metadata_manager.rs:121]: Fixed.

**Pull request merged**